packages/core-contracts/src/contracts/arks/CrossChainArk.sol

```diff
@@ -0,0 +1,403 @@
+        returns (bytes32 queueId)
+    {
+        address proxyAddress = _getTargetProxy();
+
+        // Queue a state read to get the total assets from the FleetProxy on the satellite chain
+        queueId = bridgeQueue.queueReadState(
+            satelliteChainId,
+            proxyAddress,
+            IFleetProxy.totalAssets.selector,
+            ""
+        );
+
+        emit RemoteAssetBalanceUpdateRequested(
+            queueId,
+            satelliteChainId,
+            proxyAddress
+        );
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        PUBLIC VIEW FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @inheritdoc IArk
+     * @notice Returns the total assets managed by this Ark
+     * @return assets The total balance of underlying assets held by this Ark
+     */
+    function totalAssets() public view override returns (uint256 assets) {
+        assets =
+            config.asset.balanceOf(address(this)) +
+            lastRemoteAssetBalance +
+            inflightAssets;
+    }
+
+    /**
+     * @notice Gets the target proxy address from the registry
+     * @return The target proxy address
+     * @dev Reverts if no relationship exists for the satellite chain
+     */
+    function getTargetProxy() external view returns (address) {
+        return _getTargetProxy();
+    }
+
+    /**
+     * @inheritdoc ICrossChainAssetReceiver
+     * @notice Checks if this contract supports the CrossChainReceiver interface
+     * @param interfaceId The interface ID to check
+     * @return True if the contract implements ICrossChainReceiver or ICrossChainAssetReceiver
+     */
+    function supportsInterface(
+        bytes4 interfaceId
+    )
+        external
+        pure
+        override(
+            ICrossChainAssetReceiver,
+            ICrossChainStateReadReceiver,
+            IERC165
+        )
+        returns (bool)
+    {
+        return
+            interfaceId == type(ICrossChainAssetReceiver).interfaceId ||
+            interfaceId == type(ICrossChainStateReadReceiver).interfaceId ||
+            interfaceId == type(IInflightAssetTracking).interfaceId ||
+            interfaceId == type(IERC165).interfaceId;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Boards the Ark by initiating a cross-chain transfer
+     * @param amount Amount of tokens to transfer
+     * @dev This function queues a cross-chain transfer to the target proxy using the registry
+     */
+    function _board(uint256 amount, bytes calldata) internal override {
+        address proxyAddress = _getTargetProxy();
+
+        // Approve BridgeQueue to spend tokens
+        config.asset.approve(address(bridgeQueue), amount);
+
+        bridgeQueue.queueTransferAssets(
```

Comment:
here you could get the queue id / operation id and status - to flag that there is a bridiging operation queued